src/pyhf/workspace.py

```diff
@@ -39,14 +40,23 @@ def _join_items(join, left_items, right_items, key='name'):
     joined_items = copy.deepcopy(primary_items)
     for secondary_item in secondary_items:
         # outer join: merge primary and secondary, matching where possible
+        keys = [item[key] for item in joined_items]
         if join == 'outer' and secondary_item in primary_items:
             continue
         # left/right outer join: only add secondary if existing item (by key value) is not in primary
         # NB: this will be slow for large numbers of items
-        elif join in ['left outer', 'right outer'] and secondary_item[key] in [
-            item[key] for item in joined_items
-        ]:
+        elif join in ['left outer', 'right outer'] and secondary_item[key] in keys:
+            # Deeply merge a sublist as well, if we need to
+            if deep_merge_key:
```

Comment:
```suggestion
            if deep_merge_key is not None:
```

In keeping with my `None` suggestion above.